src/Distribution/Nixpkgs/Haskell/FromCabal.hs

```diff
@@ -48,10 +49,24 @@ fromGenericPackageDescription haskellResolver nixpkgsResolver arch compiler flag
 finalizeGenericPackageDescription :: HaskellResolver -> Platform -> CompilerInfo -> FlagAssignment -> [Constraint] -> GenericPackageDescription -> (PackageDescription, [Dependency])
 finalizeGenericPackageDescription haskellResolver arch compiler flags constraints genDesc =
   let
+    -- finalizePD incooperates the 'LibraryName' of a dependency
+    -- which we always ignore, so the Cabal-compatible resolver
+    -- is a simple wrapper around our 'HaskellResolver'
+    makeCabalResolver :: HaskellResolver -> Dependency -> Bool
+    makeCabalResolver r (Dependency n v _) = r (PackageVersionConstraint n v)
+
+    -- the finalizePD API changed in Cabal 3.4.0.0, so we need to do some plumbing.
+    -- See https://github.com/haskell/cabal/issues/5570
+#if MIN_VERSION_Cabal(3,4,0)
+    makeCabalConstraints = id
```

Comment:
It might be good to have a type signature on this function.

This PR LGTM!

Reply:
Done!